Sources/ProjectSpec/AggregateTarget.swift

```diff
@@ -61,3 +62,17 @@ extension AggregateTarget: NamedJSONDictionaryConvertible {
         attributes = jsonDictionary.json(atKeyPath: "attributes") ?? [:]
     }
 }
+
+extension AggregateTarget: PathContaining {
+
+    static func expandPaths(for source: [String: JSONDictionary], relativeTo path: Path) -> [String: JSONDictionary] {
+        var result = source
+
+        for (targetName, var target) in result {
+            target = expandStringPaths(from: target, forKey: "configFiles", relativeTo: path)
+            target = expandChildPaths(from: target, forKey: "buildScripts", relativeTo: path, type: BuildScript.self)
```

Comment:
What if this `expandPaths` just returned something like `[PathProperty]` enum. Then it's a bit more declarative. And maybe `PathContainer` is good name?

Hand typed code (might not actually work):

```swift
struct PathProperty {
	let name: String
	let type: PropertyType

	enum PropertyType {
 		case string
 		case array
 		case object(PathContaining.Type) 
	}
}
```

```swift
extension AggregateTarget: PathContainer {
	static var pathProperties: [PathProperty] {
		return [
			PathProperty(name: "configFiles", type: .array),
			PathProperty(name: "buildScripts", type: .object(BuildScript.self)),
		]
	}
}
```

Then the spec loader can loop through all of these and do the actual merging

Reply:
That's a much more elegant implementation than mine (I was initially merging it with the `resolveProject(_:)` function but ended up breaking it out, so it looks similar). I'll refactor it.

Reply:
Actually generics is making this rather complicated, I'll have to give it some more thought.

> Protocol 'PathContainer' can only be used as a generic constraint because it has Self or associated type requirements

Reply:
Ahh I see. Then the .object case can just have an array of `[PathProperty]` instead of `PathContainer`

Reply:
I've managed to refactor it with just an enum:

```swift
enum PathProperty {
    case string(String)
    case dictionary([PathProperty])
    case object(String, [PathProperty])
}
```

The expansion is then handled recursively in `Project.expandPaths(_:in:relativeTo:)`.